### PR TITLE
Notes api

### PR DESCRIPTION
**Pull request by XaosSintez**

### CODE REVIEW

File: src/Entities/Note.php [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    }
+
+    /**
+     * @param string $value
+     *
+     * @return $this
+     */
+    public function setRelatedType($value)
+    {
+        return $this->setAttribute('relatedType', $value);
```
Comment (slavcodev) [inline src/Entities/Note.php]: How about validate the type if supported?

Comment (XaosSintez) [inline src/Entities/Note.php]: You mean just check in_array() ? Because we don't have much validation here

Comment (slavcodev) [inline src/Entities/Note.php]: Yeah.

File: src/Entities/Note.php [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+    /**
+     * @param string $value
+     *
+     * @return $this
+     */
+    public function setRelatedType($value)
+    {
+        if (!in_array($value, $this->allowableRelatedTypes)) {
+            throw new UnprocessableEntityException('Wrong relatedType');
```
Comment (XaosSintez) [inline src/Entities/Note.php]: Not sure, what status here is best, @slavcodev can you help please ?

Comment (adamaltman) [inline src/Entities/Note.php]: howabout `Invalid relatedType`?

Comment (slavcodev) [inline src/Entities/Note.php]: @XaosSintez please look to [this example](https://github.com/Rebilly/rebilly-php/blob/master/src/Entities/Blacklist.php#L61)

File: src/Entities/Note.php [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+ *
+ * @author Arman Tuyakbayev <arman.tuyakbayev@rebilly.com>
+ * @version 0.1
+ */
+final class Note extends Entity
+{
+    const RELATED_TYPE_CUSTOMER = 'customer';
+    const RELATED_TYPE_WEBSITE = 'website';
+
+    public $allowableRelatedTypes = [
```
Comment (slavcodev) [inline src/Entities/Note.php]: Please incapsulate this array to static method.

**Pull request merged**